### PR TITLE
[synthetics] Fix the reporting of residual results

### PR DESCRIPTION
**Pull request by Drarig29**
### What and why?

💡 _Better reviewed commit by commit_

Our unit tests were incomplete and #1097 broke something when we have residual results, i.e. when datadog-ci stops polling before our backend.

We can reproduce this edge case with a really small value of `--pollingTimeout`, like 1 millisecond:

**Expected:**

image

**Actual:**

image

### How?

- Fix the unit tests: 1ef26d01e3f5e04d677645960b35bfb819929f93 and 30d211fc8a3e42991e9c8e3bb88686306dee40dc
- Fix the code: 5e40fc51c918f82ea0ac39ec18eca2082da6f7c1 and a93c75a79390dde14ab0b0709a69e5b3d4b5153d

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)

### CODE REVIEW

**Review by etnbrd — APPROVED** [review]
LGTM 👍

File: src/commands/synthetics/utils.ts [hunk 1]
```diff
@@ -402,7 +410,7 @@ const reportResidualResults = (
 
   const notEmittedResultIds = notEmitted.map((r) => r.result_id)
 
-  reportResults(pollResultMap, notEmittedResultIds, batch, resultDisplayInfo, false, reporter)
+  reportResults(pollResultMap, notEmittedResultIds, batch, resultDisplayInfo, hasBatchExceededMaxPollingDate, reporter)
```
[the remark this answers is not in the document]
Comment (Drarig29) [reply src/commands/synthetics/utils.ts:413]: Created a follow-up card 👍

[the remark this answers is not in the document]
Comment (Drarig29) [reply src/commands/synthetics/utils.ts]: I did it in this PR directly

**Review by etnbrd — APPROVED** [review]
LGTM 👍
